decks/node-beginner/meeting-2.mdx

```diff
@@ -0,0 +1,225 @@
+</p>
+
+</CodeLayout>
+
+---
+
+# Exercises 💪🏼
+
+_To be completed during the meeting_
+
+Time to practice what you have learned from the readings.
+
+---
+
+# Exercise 1
+
+Hit the endpoint that returns your user information (username, location, avatar URL, etc.)
+
+<p style={{fontSize: '0.55em', fontStyle: 'italic'}}>
+Use this documentation as a reference for all questions https://developer.github.com/v3
+</p>
+
+<MinuteTimer minutes={5} />
+
+---
+
+# Exercise 1 - Answer
+
+GET https://api.github.com/user
+
+---
+
+# Exercise 2
+
+Ask the person to your right for their GitHub username. Use that information to get their user information (username, location, avatar URL, etc.)
+
+<MinuteTimer minutes={5} />
+
+---
+
+# Exercise 2 - Answer
+
+GET https://api.github.com/users/:username
+
+---
+
+# Exercise 3
+
+1. List all your repositories that you are either a collaborator or owner of.
+2. Sort it by the created at date in ascending order.
+
+<MinuteTimer minutes={5} />
+
+---
+
+# Exercise 3 - Answer
+
+1. GET https://api.github.com/user/repos?affiliation=owner,collaborator
+2. GET https://api.github.com/user/repos?affiliation=owner,collaborator&sort=created&direction=asc
+
+---
+
+# Exercise 4
+
+1. Get a list of all repositories of person next to you.
+2. Sort by create date in descending order.
+
+<MinuteTimer minutes={5} />
+
+---
+
+# Exercise 4 - Answer
+
+1. GET https://api.github.com/users/:username/repos
+2. GET https://api.github.com/users/:username/repos?sort=created&direction=desc
+
+---
+
+# Exercise 5
+
+Using a repository from the solution above, check if that repository has vulnerability alerts enabled
+
+_Note: you need to include `application/vnd.github.dorian-preview+json` in the `Accept` header since this feature is still in preview._
+
+Need help? https://developer.github.com/v3/repos/#enable-vulnerability-alerts
+
+<MinuteTimer minutes={5} />
+
+---
+
+# Exercise 5 - Answer
+
+- Check if vulnerability alerts are enabled for a repository: GET https://api.github.com/repos/:owner/:repo_name/vulnerability-alerts
+  - If Repo is enabled you should see `Status: 204 No Content` in response headers
+  - If Repo is NOT enabled you should see `Status: 404 Not Found` in response headers
+- Enable vulnerability alerts for a repo: PUT https://api.github.com/repos/:owner/:repo_name/vulnerability-alerts
+- Disable vulnerability alerts for a repo: DELETE https://api.github.com/repos/:owner/:repo_name/vulnerability-alerts
+
+---
+
+# Exercise 6
+
+- Get a list of all commits with either the words `delete` OR `ignore`
+- Assuming 75 results per page, return page 3
+
+_Note: you need to include `application/vnd.github.cloak-preview` in the `Accept` header since this feature is still in preview._
+
+Need help? https://developer.github.com/v3/search/#search-commits
+
+<MinuteTimer minutes={5} />
+
+---
+
+# Exercise 6 - Answer
+
+- GET https://api.github.com/search/commits?q='delete OR ignore'
```

Comment:
We switch from numbered list to bullet list for the answers slide. Lets keep it consistent across all slides.

Reply:
I think it's numbered list for question and bullet for answer?

Reply:
There's an odd formatting in the query string and how it shows up in Markdown.